src/tools/rbd_cache/ObjectCacheStore.cc

```diff
@@ -63,13 +63,13 @@ int ObjectCacheStore::do_promote(std::string pool_name, std::string object_name)
     m_cache_table.emplace(cache_file_name, PROMOTING);
   }
 
-  librados::bufferlist read_buf;      
+  librados::bufferlist* read_buf = new librados::bufferlist();
```

Comment:
Memory leak: There's no `delete` for `read_buf`

Reply:
@mingnus We have a new commit with will do async read here, will include this fix, thanks!